### ISSUE TITLE
@Clerk/remix broken SSR

### ISSUE THREAD

**Issue by jtparrett** [clerkinc/javascript#171]
## Package + Version
- [x] `@clerk/remix`

### Version:
```
0.3.5
```

## Description
After upgrading to version 0.3.5 of clerk/remix SSR seems to no longer work.

Everything has been configured per the clerk/remix setup instructions.
Here is my root loader (note loadUser):
```
export const loader: LoaderFunction = (args) =>
  rootAuthLoader(
    args,
    {
      loadUser: true,
    }
  )
```
  
Here's my page:
```
const Page = () => {
  const auth = useAuth()
  
  console.log(auth)
  
  if(auth.isSignedIn){
    return <p>Some user!</p>
  }
  
  return <p>No user!</p>
}
```

**On the server** the log is like so:
```
{
  isLoaded: true,
  isSignedIn: false,
  sessionId: null,
  userId: null,
  signOut: [AsyncFunction (anonymous)],
  getToken: [AsyncFunction (anonymous)]
}
```

**On the client** the log is like so:
```
isLoaded: true
isSignedIn: true
sessionId: "..."
userId: "..."
signOut: async (...args) => {…}
getToken: async (options) => {…}
```

The end result for the user is a flash on page load which shows "No user", and then it switches to "Some user!" once the client has finished loading.

**Comment by nikosdouvlis** [comment]
Hello @jtparrett, thanks for the report. My first guess is that this is an env variable mismatch Can you check that both the `CLERK_FRONTEND_API` and `CLERK_API_KEY` environment variables point to the same instance? 

Some more questions that could help:
1. Which version were you using before? 
2. Is this happening locally or when deployed? If it's the latter, where is the app deployed to?
3. Is there a live URL we can take a look at?

**Comment by jtparrett** [comment]
@nikosdouvlis thanks for your response! 

I checked all my keys and they seemed to be in order.

1. I was using 0.3.4 before.
2. It was happening locally and in production.
3. Unfortunately I can't grant you access to the project as it's a closed beta (I might be able to spin up a fake project to replicate if I get time today)

Update: I cleared all my browser data (including 3rd party cookies), then re-authenticated and the problem seemed to go away. So is it possible the cookie/localstorage stuff had gotten out of sync during the update?

**Comment by colinclerk** [comment]
I just saw your message on Twitter:
> I raised the original issue 😁 thank you so much! It’s definitely related to a change that happened in 0.3.5 as I can downgrade to 0.3.4 and the issues goes away. Will try 0.3.6 later today!

Thank you for the additional context!

**Comment by jtparrett** [comment]
Hi @colinclerk , thanks for the response!
Sorry for hijacking your marketing tweet.

Yes, I am running production and staging from the same domain 😄 
Is there somewhere I can keep updated about the progress of this bug?

I've also just tried to create a fresh application and was not able to replicate the problem even when switching between 0.3.4 and 0.3.5 which makes me think it must be the domain issue! 

I've also just upgraded to 0.3.6 (latest) and upgraded to the latest remix version in my main application and i'm not seeing the original problem at all 🤞 so I'll close this issue for now, thanks again so much for getting back to me!
